### PR TITLE
Fyle 1x64m5r : Adding shimmers to my-advances page

### PR DESCRIPTION
**Pull request by siddharth0-0**
`<<< Advances  >>>`

https://user-images.githubusercontent.com/66034160/147456140-10e77182-1196-4dc8-b668-f89b300dd68f.mp4

https://user-images.githubusercontent.com/66034160/147456145-d824aa41-5b8f-4514-90d7-21e9cb304b0e.mp4

`<<< Team Advances  >>>`

https://user-images.githubusercontent.com/66034160/148214308-1dff35f9-3b80-4fff-a1e5-ef19307b988b.mp4

https://user-images.githubusercontent.com/66034160/148214334-49176cef-ebc1-4a39-9720-baf3966c939e.mp4

### CODE REVIEW

File: src/app/fyle/my-advances/my-advances.page.html [hunk 1]
```diff
@@ -36,14 +36,20 @@
     </div>
   </ng-container>
 
+  <app-fy-loading-screen
+    *ngIf="isLoading"
+    [isSelectionModeEnabled]="selectionMode"
+    class="my-expenses--shimmers"
```
Comment (Dimple16) [inline src/app/fyle/my-advances/my-advances.page.html:42]: check if this class is required, this isn't added in CSS file, can be removed

File: src/app/fyle/my-advances/my-advances.page.ts [hunk 2]
```diff
@@ -54,7 +54,7 @@ export class MyAdvancesPage {
 
   navigateBack = false;
 
-  refreshAdvances$: Subject<void> = new Subject();
+  refreshAdvances$ = new BehaviorSubject<void>(null);
```
Comment (Dimple16) [inline src/app/fyle/my-advances/my-advances.page.ts:57]: intentional change?

**Review by Dimple16 — CHANGES_REQUESTED** [review]
we should make similar changes in Team advances page too

File: src/app/fyle/my-advances/my-advances.page.html [hunk 3]
```diff
@@ -36,13 +36,15 @@
...
+
   <div class="my-advances--body" [ngClass]="{'my-advances--zero-state-body': (advances$|async)?.length === 0 }">
     <ion-refresher slot="fixed" (ionRefresh)="doRefresh($event)">
       <ion-refresher-content></ion-refresher-content>
     </ion-refresher>
 
-    <ng-container *ngIf="advances$|async as advances">
-      <ng-container *ngIf="advances?.length === 0">
+    <ng-container *ngIf="!isLoading">
+      <ng-container *ngIf="(advances$ | async)?.length===0">
```
Comment (Dimple16) [inline src/app/fyle/my-advances/my-advances.page.html:47]: add spacing

**Review by Dimple16 — CHANGES_REQUESTED** [review]
this has team advances page changes too, can you add a loom?

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/team-advance/team-advance.page.html [hunk 4]
```diff
@@ -42,18 +42,22 @@
...
-    <div *ngFor="let teamAdvancerequest of teamAdvancerequests$ | async as list; let i=index">
-      <app-team-adv-card
-        [advanceRequest]="teamAdvancerequest"
-        [prevDate]="list[i-1]?.areq_created_at"
-        (gotoAdvance)="onAdvanceClick($event)"
-      >
-      </app-team-adv-card>
+    <app-fy-loading-screen *ngIf="isLoading"></app-fy-loading-screen>
+
+    <div *ngIf="!isLoading">
```
Comment (abhishek1234321) [inline src/app/fyle/team-advance/team-advance.page.html:47]: same comments as above

File: src/app/fyle/team-advance/team-advance.page.ts [hunk 5]
```diff
@@ -110,6 +108,10 @@ export class TeamAdvancePage implements OnInit {
     this.count$.subscribe(noop);
     this.isInfiniteScrollRequired$.subscribe(noop);
     this.loadData$.next({ pageNumber: this.currentPageNumber, state: this.state });
+
+    setTimeout(() => {
```
Comment (abhishek1234321) [inline src/app/fyle/team-advance/team-advance.page.ts:112]: same comments as above

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: src/app/fyle/my-advances/my-advances.page.ts [hunk 6]
```diff
@@ -66,6 +66,8 @@ export class MyAdvancesPage {
 
   filterParams$ = new BehaviorSubject<Filters>({});
 
+  isLoading = false;
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/fyle/my-advances/my-advances.page.ts:69]: we can skip this for boolean, number - https://palantir.github.io/tslint/rules/no-inferrable-types/ 
inferred directly, there is lint added too - `Type boolean trivially inferred from a boolean literal, remove type annotation` cc @abhishek1234321

File: src/app/fyle/my-advances/my-advances.page.ts [hunk 7]
```diff
@@ -154,57 +158,55 @@ export class MyAdvancesPage {
...
+              });
+            }
+
+            newArr = this.utilityService.sortAllAdvances(filters.sortDir, filters.sortParam, newArr);
+            return newArr;
+          })
         )
       )
     );
+    setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (siddharth0-0) [reply src/app/fyle/my-advances/my-advances.page.ts:207]: 500ms is just been added to load the shimmers.

[the remark this answers is not in the document]
Comment (abhishek1234321) [reply src/app/fyle/my-advances/my-advances.page.ts:207]: @Dimple16 could you check this? ideally we should hide shimmers only after the data is loaded

**Review by iravisankar — APPROVED** [review]

**Comment by iravisankar** [comment]
@siddharth0-0  please confirm shimmers on infinite scroll

**Review by abhishek1234321 — APPROVED** [review]

**Pull request merged**